crates/sp-executor/src/lib.rs

```diff
@@ -296,6 +296,9 @@ pub enum VerificationError {
     StorageRootDecode(parity_scale_codec::Error),
     /// Failed to decode the header produced by `finalize_block`.
     HeaderDecode(parity_scale_codec::Error),
+    /// Runtime api error.
+    #[cfg(feature = "std")]
```

Comment:
Are you sure `VerificationError` enum should even be in this crate anymore? It doesn't seem to belong here if we're not doing verification in the runtime.

Reply:
I thought that too initially, but still found one usage:
https://github.com/subspace/subspace/blob/961104c4b5a846799adf17bd5d8af949d912d19c/crates/sp-executor/src/lib.rs#L266-L269

`decode_execution_result` is now only used in the client, perhaps we can make it std only in sp-executor, then whole `VerificationError` can be std only. I don't have a strong preference as I don't think it's siginificant.

Reply:
Okay, another reason to keep it in this crate https://github.com/subspace/subspace/pull/551/commits/961104c4b5a846799adf17bd5d8af949d912d19c#diff-c7fab4838df44656cad8198d607fd03699ad595255a9a5b767ed9487291f9322R253-R255 . Then the problem becomes if we want to make this error enum std only.

Reply:
The question then is whether we should try to smash everything into this enum or we can have several separate enums and a way to convert one into another if necessary.

Reply:
I don't quite get your last comment, what do you mean by several separate enums?

Reply:
If there are different places that generate errors, then we can have multiple enums like this for errors. Even if they are similar, that is fine, we don't have to reuse enums if it is not a good fit.